src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -991,6 +998,237 @@ def progress_callback(current, total):
+        self.image = image
+        self.target_port = target_port
+        self.is_main = is_main
+        self.start_up_command = start_up_command
+        self.auth_type = auth_type
+        self.user_name = user_name
+        self.password_secret = password_secret
+        self.user_managed_identity_client_id = user_managed_identity_client_id
+        self.volume_mounts = volume_mounts
+        self.environment_variables = environment_variables
+
+    @classmethod
+    def from_json(cls, json_data):
+        name = json_data.get("name")
+        properties = json_data.get("properties", {})
+        volume_mounts = properties.get("volumeMounts")
+        if volume_mounts:
+            for mount in volume_mounts:
+                if "containerMountPath" in mount:
+                    mount["container_mount_path"] = mount.pop("containerMountPath")
+                if "volumeSubPath" in mount:
+                    mount["volume_sub_path"] = mount.pop("volumeSubPath")
+                if "readOnly" in mount:
+                    mount["read_only"] = mount.pop("readOnly")
+        return cls(
+            name=name,
+            image=properties.get("image"),
+            target_port=properties.get("targetPort"),
+            is_main=properties.get("isMain"),
+            start_up_command=properties.get("startUpCommand"),
+            auth_type=properties.get("authType"),
+            user_name=properties.get("userName"),
+            password_secret=properties.get("passwordSecret"),
+            user_managed_identity_client_id=properties.get("userManagedIdentityClientId"),
+            volume_mounts=volume_mounts,
+            environment_variables=properties.get("environmentVariables")
+        )
+
+
+def create_webapp_sitecontainers(cmd, name, resource_group, container_name=None, image=None, target_port=None,
+                                 slot=None, startup_cmd=None, is_main=None, system_assigned_identity=None,
+                                 user_assigned_identity=None, registry_username=None,
+                                 registry_password=None, sitecontainers_spec_file=None):
+    import os
+    # verify if site is a linux site
+    client = web_client_factory(cmd.cli_ctx)
+    app = client.web_apps.get(resource_group, name)
+    if not is_linux_webapp(app):
+        raise ValidationError("Site is not a linux webapp. Sitecontainers are only supported for linux webapps.")
+    response = None
+
+    # check if sitecontainers_spec_file is provided
+    if sitecontainers_spec_file:
+        response = []
+        logger.warning("Using sitecontainer spec file to create sitecontainer(s)")
+        if not os.path.exists(sitecontainers_spec_file):
+            raise ValidationError("The sitecontainer spec file does not exist at the path '{}'"
+                                  .format(sitecontainers_spec_file))
+        with open(sitecontainers_spec_file, 'r') as file:
+            sitecontainers_spec_json = json.load(file)
```

Comment:
This will throw an exception if json is not valid. Lets catch and send appropriate error message.